src/utils/url.js

```diff
@@ -95,7 +95,11 @@ export const getNarrowFromLink = (url: string, realm: string, users: any[]): Nar
       decodeURIComponent(transformToEncodedURI(paths[3])),
     );
   } else if (isStreamLink(url, realm)) {
-    return streamNarrow(decodeURIComponent(transformToEncodedURI(paths[1])));
+    const decoded = decodeURIComponent(transformToEncodedURI(paths[1]));
+    // New url format: #narrow/stream/{stream_id}-{stream_name}. We need to extract the stream_name.
+    const index = decoded.indexOf('-') === -1 ? 0 : decoded.indexOf('-') + 1;
```

Comment:
That is likely more clearly expressed as a regular expression.

Would be good to add a unit-test to prove the correctness of this. (The unit test should be failing without your change.)

Reply:
Create something like util function and extract this logic from here. Because it is going to be re-used for topic links too (e.g.: https://chat.zulip.org/#narrow/stream/48-mobile/topic/writing.20tests)

Also in the conversation link https://chat.zulip.org/#narrow/stream/48-mobile/subject/writing.20tests/near/560084

Test for other links too.

Reply:
Wrote tests, refactored the logic. Ready to go in 🏃🏻‍♂️  @borisyankov @jainkuniya